syntax = "proto3";

package nighthawk.server;

import "google/protobuf/wrappers.proto";
import "validate/validate.proto";
import "envoy/api/v2/core/base.proto";
import "google/protobuf/duration.proto";

message GaugeBasedDelay {
  // Name of the guage providing the target value.
  string guage_name = 1;
  int64 guage_target_value = 2;
  // The delay is computed as (guauge_target_value - guage_value) * delta_delay;
  google.protobuf.Duration delay_delta = 3 [(validate.rules).duration.gte.nanos = 0];
}

// Options that control the test server response.
message ResponseOptions {
  // List of additional response headers.
  repeated envoy.api.v2.core.HeaderValueOption response_headers = 1;
  // Number of 'a' characters in the the response body.
  uint32 response_body_size = 2 [(validate.rules).uint32 = {lte: 4194304}];
  // If true, then echo request headers in the response body.
  bool echo_request_headers = 3;

  oneof oneof_delay_options {
    // Static delay duration.
    google.protobuf.Duration static_delay = 4 [(validate.rules).duration.gte.nanos = 0];
    // Guage based delay configuration.
    GaugeBasedDelay stats_based_delay = 5;
  }
}
